operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +64,94 @@ type MachineConfigurationList struct {
 	// Items contains the items
 	Items []MachineConfiguration `json:"items"`
 }
+
+type ManagedBootImages struct {
+	// machineManagers is an array of machineManager objects.
+	// The MCO will watch for changes to this list and enroll/un-enroll machine management resources. An entry in this list
+	// consists of the resource type, the API group that the resource belongs to and a selection filter on the resources.
+	// All fields are required.
+	//
+	// Warning: Only one entry per machine management resource type is allowed. If multiple entries target the same type of machine management
+	// resource, all entries associated with that resource type will be ignored. For example, if one entry describes that MAPI machinesets are
+	// disabled for boot image updates and a later entry describes that MAPI machinesets are enabled for boot image updates, both entries
+	// will be ignored.
+	//
+	// +kubebuilder:validation:Required
+	MachineManagers []MachineManager `json:"machineManagers"`
+}
+
+// MachineManager contains identifying information of a machine management resource(eg. a machineset) that will be
+// enrolled for boot image updates. This is likely to evolve as support for more machine management resources are added.
+type MachineManager struct {
+	// resource is the machine management resource's type.
+	//
+	// The following values are accepted:
+	// - MachineSets: The machine manager will only enroll resources of the type MachineSet, which may belong to MachineAPI or ClusterAPI.
+	//
+	// +kubebuilder:validation:Required
+	Resource MachineManagerManagedResourceType `json:"resource"`
+	// apiGroup is name of the APIGroup that the machine management resource belongs to.
+	//
+	// The following values are accepted:
+	// - MachineAPI: The machine manager will only enroll resources that belong to MachineAPI APIGroup.
+	//
+	// +kubebuilder:validation:Required
+	APIGroup MachineManagerManagedAPIGroupType `json:"apiGroup"`
+	// selection allows granular control of the machine management resources that will be enrolled to the ManagedBootImages feature.
+	// +kubebuilder:validation:Required
+	Selection MachineManagerSelector `json:"selection"`
+}
+
+// +kubebuilder:validation:XValidation:rule="has(self.mode) && self.mode == 'Partial' ?  has(self.partial) : !has(self.partial)",message="Partial is required when type is partial, and forbidden otherwise"
+// +union
+type MachineManagerSelector struct {
+	// mode is a union discriminator for MachineManagerSelector and can have three possible values.
+	// - All: All machinesets are enrolled into the ManagedBootImages feature.
```

Comment:
I would prefer not to describe the feature by the name of the feature, describe it instead by the behaviour. Right now, I have to know what ManagedBootImages feature is, you could instead talk about having automatically updated boot images during cluster upgrades, and then I wouldn't need to understand the feature name